service-api/module/Application/src/Controller/YotiController.php

```diff
@@ -56,22 +64,37 @@ public function findPostOfficeAction(): JsonModel
         $this->getResponse()->setStatusCode(Response::STATUS_CODE_200);
         return new JsonModel($branches);
     }
-    public function createSessionAction(array $sessionData): JsonModel
+    public function createSessionAction(): JsonModel
     {
         $uuid = $this->params()->fromRoute('uuid');
-        //@TODO authenticate if not using mock?
-        $result = $this->yotiService->createSession($sessionData);
-        //save sessionId back to caseData
-        if ($result["status"] < 400) {
-            $this->dataImportHandler->updateCaseData(
-                $uuid,
-                'sessionId',
-                'S',
-                $result["data"]["session_id"]
-            );
+
+        if (! $uuid) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return new JsonModel(['error' => 'Missing uuid']);
         }
-        $this->getResponse()->setStatusCode(Response::STATUS_CODE_204);
-        return new JsonModel(null);
+        $sessionUuid = strval(Uuid::uuid4());
+        $caseData = $this->dataQuery->getCaseByUUID($uuid);
+        $sessionData = $this->sessionConfig->build($caseData, $sessionUuid);
```

Comment:
`$authToken` is a clearer description of what the UID is for
```suggestion
        $authToken = strval(Uuid::uuid4());
        $caseData = $this->dataQuery->getCaseByUUID($uuid);
        $sessionData = $this->sessionConfig->build($caseData, $authToken);
```